templates/agent/scripts/couchbase.py.j2

```diff
@@ -0,0 +1,158 @@
+    """
+    keys = keys_string.split(".")
+    for key in keys:
+        data = data[key]
+    return data
+
+
+def handle_cluster(args):
+    # couchbase['cluster', key, type]
+    data = get_from_url(cluster_url)
+    data_type = args[1]
+    keys_string = args[0]
+
+    result = get_data_from_keys(data, keys_string)
+
+    if keys_string == "nodes":
+        result = [item['hostname'].split(":")[0] for item in result]
+        print handle_data_by_type(result, data_type, "#NODEHOST")
+        exit(0)
+
+    print handle_data_by_type(result, data_type)
+
+
+def handle_bucket(args):
+    # warning: you can not call your bucket name to "names",
+    # it will conflict with the following condition.
+    # couchbase['bucket', 'names', type]
+    # type here is `string_list`
+    if len(args) == 2 and args[0] == 'names':
+        data = get_from_url(buckets_url)
+        bucket_names = [bucket['name'] for bucket in data]
+        print handle_data_by_type(bucket_names, args[1], "#BUCKETNAME")
+        exit(0)
+
+    # couchbase['bucket', name, key, type]
+    name = args[0]
+    data_type = args[2]
+
+    bucket_url = urlparse.urljoin(buckets_url, "%s/" % name)
+
+    # couchbase['bucket', name, 'stats.*', type]
+    keys_string = args[1]
+    if keys_string.startswith("stats"):
+        bucket_stats_url = urlparse.urljoin(bucket_url, "stats?zoom=minute")
+        data = get_from_url(bucket_stats_url)
+        result = get_data_from_keys(data, keys_string.split(".", 1)[1])
+    else:
+        data = get_from_url(bucket_url)
+        # couchbase['bucket', name, "nodes.hostnames", "string"]
+        if keys_string == "nodes.hostnames":
+            result = [node['hostname'] for node in get_data_from_keys(data, "nodes")]
+            result = ", ".join(result)
+        # couchbase['bucket', name, "*", type]
+        else:
+            result = get_data_from_keys(data, keys_string)
+
+    print handle_data_by_type(result, data_type)
+
+
+def handle_node(args):
+    # couchbase['node', hostname, key, type]
+    hostname = args[0]
+    nodes = get_from_url(cluster_url).get("nodes")
+    data_type = args[2]
+    for node in nodes:
+        if node.get("hostname").split(":")[0] == hostname:
+            data = node
+            break
+    result = get_data_from_keys(data, args[1])
+
+    print handle_data_by_type(result, data_type)
+
+
+def main():
+    args = sys.argv[1:]
+
+    if len(args) < 1:
+        print "argument not provided."
+        exit(1)
+
+    # The object_type (first agrument) is either cluster, bucket, or node.
+    object_type = args[0]
+    extra_args = [arg for arg in args[1:] if arg != ""]
+
+    if object_type == "cluster":
+        handle_cluster(extra_args)
+
+    elif object_type == "bucket":
+        handle_bucket(extra_args)
+
+    elif object_type == "node":
+        handle_node(extra_args)
+    else:
+        print "Argument is not either cluster, bucket, node"
+        exit(1)
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
You probably want to have a clean and clear failure rather than a crash in error.

See https://github.com/wiredcraft-ops/role-wcl-zabbix-scripts/blob/master/templates/agent/scripts/haproxy.py.j2#L95-L120 as example